File: app/modules/account_groups/schemas.py
from __future__ import annotations

from datetime import datetime

from pydantic import Field

from app.modules.shared.schemas import DashboardModel


class AccountGroupUpsertRequest(DashboardModel):
    name: str
    account_ids: list[str] = Field(default_factory=list)


class AccountGroupResponse(DashboardModel):
    id: str
    name: str
    account_ids: list[str] = Field(default_factory=list)
    account_count: int
    created_at: datetime
    updated_at: datetime


class AccountGroupsResponse(DashboardModel):
    groups: list[AccountGroupResponse] = Field(default_factory=list)


class AccountGroupDeleteResponse(DashboardModel):
    status: str

File: frontend/src/features/account-groups/schemas.ts
import { z } from "zod";

export const AccountGroupSchema = z.object({
  id: z.string(),
  name: z.string(),
  accountIds: z.array(z.string()),
  accountCount: z.number().int().nonnegative(),
  createdAt: z.string().datetime({ offset: true }),
  updatedAt: z.string().datetime({ offset: true }),
});

export const AccountGroupsResponseSchema = z.object({
  groups: z.array(AccountGroupSchema),
});

export const AccountGroupUpsertSchema = z.object({
  name: z.string().min(1),
  accountIds: z.array(z.string()).default([]),
});

export const AccountGroupDeleteResponseSchema = z.object({
  status: z.string(),
});

export type AccountGroup = z.infer<typeof AccountGroupSchema>;
export type AccountGroupsResponse = z.infer<typeof AccountGroupsResponseSchema>;
export type AccountGroupUpsert = z.infer<typeof AccountGroupUpsertSchema>;
